### ISSUE TITLE
focus issue on single select - holding mouse button on open fixes it?

### ISSUE THREAD

**Issue by morrow95** [select2/select2#3865]
```
<select id="vinfo-timezone" class="form-control select2" name="settings[timezone]" style="width:100%;">
	<?php get_timezone($result['timezone']); ?>
</select>
```

Using the above to populate all timezones so a user can select.  What happens is on mouse click the dropdown opens, but it is similar to the mouse button being held down even when it is not... I can move the cursor around and all the text/content on the page is highlighted as if I were holding the button down for a copy.  Upon another click of the mouse the 'highlighting' is removed, but it also closes the dropdown making the whole select unusable.

The strange behavior I found is upon initial click of the select if I hold my mouse button down until the drop down shows, then release it, everything works correctly.  I did some digging and I believe this could be because of a settimeout() in the actual select2 code for single selects, but I don't see how it would affect this situation.

The php function is going to run on load so all the options would present before anyone has a chance to see/click on the select.  Is it simply because the list is rather large (all the timezones ```foreach(DateTimeZone::listIdentifiers() as $tz) {...```) or what?

**Comment by morrow95** [comment]
I played around with this some more... if I manually enter options in my html (not using php to populate them) the issue still occurs, but only seems to happen if the list of options is rather large.  When only a few options are entered it does not occur.  This eliminates php being a problem so it seems as though this has something to do with the size of the options?

I would still like to know why holding the mouse down for around on second or so on initial click ALWAYS prevents this and it works as it should.

I love select2, but this is the second time around I have tried integrating 4.0 into our project and it has been nothing but problems whereas 3.5 has been working great for awhile now with none of these issues and the exact same code (minus the select2 initialization changes needed).

**Comment by kevin-brown** [comment]
If we could get a jsbin that consistently reproduced this, that would give us a decent starting place. It _may_ be an issue with large data sets, but usually there aren't enough timezones to cause problems.

---

Can you expand upon the issue you are facing by adding a few more pieces of information to your bug report?

- What steps can be used to reproduce the issue?
- What is the bug and what is the expected outcome?
- What browser(s) and Operating System have you tested with?
- Does the bug happen consistently across all tested browsers?
- What version of jQuery are you using? And what version of Select2?
- Are you using Select2 with other plugins?

The [contributing guide][cg] contains a useful section with a few questions which can help us track down the bug and speed up the process of finding a fix for it.  The creation of [an isolated test case][test-case] would also be useful.

[cg]: https://github.com/select2/select2/blob/master/CONTRIBUTING.md#reporting-bugs-with-select2
[test-case]: http://css-tricks.com/6263-reduced-test-cases/

**Comment by morrow95** [comment]
Just loaded rc 4.0.1 and this issue has disappeared.  Not sure exactly why, but it no longer happens.

**Comment by kevin-brown** [comment]
> Just loaded rc 4.0.1 and this issue has disappeared.

Interesting, we had a few focus issues that were fixed with 4.0.1-rc.1, so that may be related.

If this comes up again, we can reopen this ticket.

**Comment by morrow95** [comment]
Sounds good... I was able to reproduce this consistently, but since loading the rc it has not happened once.  Nothing has changed other than the s2 files so whatever was causing it must have been related somehow.
